include/simdjson/dom/parser.h

```diff
@@ -391,6 +366,11 @@ class parser {
   /** @private Private and deprecated: use `parser.parse(...).doc.dump_raw_tape()` instead */
   inline bool dump_raw_tape(std::ostream &os) const noexcept;
 
+  /** @private Number of structural indices passed from stage 1 to stage 2 */
+  uint32_t& n_structural_indexes() const noexcept;
+  /** @private Structural indices passed from stage 1 to stage 2 */
+  std::unique_ptr<uint32_t[]>& structural_indexes() const noexcept;
```

Comment:
Is there no way to avoid this indirection? It forces you to add "()" throughout which harms the readability of the code (subjectively speaking).

Reply:
Sure, I can make them access parser.implementation.*. I intend to change it up down the road so that the benchmarkers use an official interface and document stream doesn't need them at all, but no reason to make them suffer in the meantime.

Reply:
On that note, too: I have noticed that accessing members through indirection can actually eliminate certain optimizations. In particular, the `array_ptr[index++]` optimization seems like it gets easily tripped up by even a `really_inline` indirection.

Reply:
Removed the indirection (now it does parser.implementation->structural_indexes).